scripts/require-sync/main.go

```diff
@@ -0,0 +1,193 @@
+// Verify referential templates include sync data
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	opa "github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/drivers/local"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/core/templates"
+	"github.com/open-policy-agent/gatekeeper/pkg/target"
+	"gopkg.in/yaml.v3"
+)
+
+const syncAnnotation string = "metadata.gatekeeper.sh/requiresSyncData"
+
+var (
+	pathFlag = flag.String("path", "", "Path to verify referential templates include sync data.")
+	fileFlag = flag.Bool("sync_file", false, "When `true`, require a `sync.yaml` file for each referential template.")
+)
+
+func main() {
+	flag.Parse()
+	if *pathFlag == "" {
+		log.Fatal("Missing `path` flag")
+	}
+	log.Println("Verifying path:", *pathFlag)
+
+	err := checkTemplates(*pathFlag)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func checkTemplates(libraryPath string) error {
+	err := filepath.WalkDir(libraryPath, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if !d.IsDir() {
+			return nil
+		}
+
+		_, error := os.Stat(filepath.Join(path, "template.yaml"))
+		if !os.IsNotExist(error) {
+			// Unmarshall template
+			krm, err := unMarshall(filepath.Join(path, "template.yaml"))
+			if err != nil {
+				return err
+			}
+
+			// Check if Template is Referential
+			isRef, err := isReferential(krm.ConstraintTemplate)
+			if err != nil {
+				return fmt.Errorf("detecting referential: %w", err)
+			}
+			if isRef {
+				// Get template name
+				templateName, err := getTemplateName(krm.ConstraintTemplate)
+				if err != nil {
+					return err
+				}
+				log.Println(fmt.Errorf("Found Referential Template: %s", templateName))
```

Comment:
is this an error? Should we use `fmt.Sprintf()`?

Reply:
Hi @maxsmythe  - I believe that's from an earlier commit, it shouldn't be in the current version: https://github.com/open-policy-agent/gatekeeper-library/pull/251/files#diff-f987eb96fe4bcda2fea78c6a08551b7947c935c4a3a4d5ce8af88083957e9af2R96